Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/audit/transform/AuditTransforms.java

```diff
@@ -135,11 +142,16 @@ public final LogEventRequestType transformResponseToAuditMsg(T request, K respon
         AuditMessageType auditMsg = createBaseAuditMessage(assertion, target, isRequesting, webContextProperties,
             serviceName);
         auditMsg = getParticipantObjectIdentificationForResponse(request, response, assertion, auditMsg);
-
+        String userID = getUserId(auditMsg.getActiveParticipant());
+        AssertionHelper assertionHelper = new AssertionHelper();
+        if (StringUtils.isEmpty(userID)){
+            userID = assertionHelper.extractUserName(assertion);
```

Comment:
This fix only apply for inbound message. The audit retrieves correctly for outbound message. I will address assertionHelper inline method in the next commit.